### PR TITLE
Fixed #29426 -- Made UUID inputs in the admin match the width of a UUID.

### PR DESCRIPTION
**Pull request by mackong**
https://code.djangoproject.com/ticket/29426

### CODE REVIEW

File: django/forms/fields.py [hunk 1]
```diff
@@ -1189,7 +1189,7 @@ class UUIDField(CharField):
 
     def prepare_value(self, value):
         if isinstance(value, uuid.UUID):
-            return value.hex
```
Comment (aaugustin) [inline django/forms/fields.py]: This lines dates back to the original UUID patch in ed7821231b7dbf34a6c8ca65be3b9bcbda4a0703.

I didn't spot anything in https://code.djangoproject.com/ticket/19463 discussing this.

This change change looks reasonably safe to me but may be worth a line in the release notes.

Comment (mackong) [reply django/forms/fields.py]: Now, `show_with_dashes` argument is added, make it more safety.

**Comment by claudep** [comment]
What would be the issue with always showing the dashes?

**Comment by aaugustin** [comment]
The assumption here isn't that there's only the hex representation.

It is that the hex representation is the most common and those who wish a different representation can subclass the field.

I suggested to default to the representation with dashes in form fields because it's the most common for human consumption (insofar as UUID can be considered suitable for human consumption).

Unless I missed it, no one brought up a reason to prefer another representation in form fields.

**Comment by mackong** [comment]
hi @codingjoe. the hex format here is just for display in form fields, and the value of UUIDField is a `uuid.UUID` instance, `bytes`, `int` of `uuid.UUID` is there, so no other format should needed.

**Comment by codingjoe** [comment]
Ok, I see your points. It was just a suggestion really, maybe our future machine overlords prefer binary ;)
The dashes definitely help to find the version bits. It would be actually cool if you could generate a v4 uuid via javascript.

**Comment by timgraham** [comment]
I created https://github.com/django/django/pull/10311 to always render UUIDs with dashes. I think that's fine rather than having a form field argument to control the behavior.

As for the width of the admin input, I think we should use a new `vUUIDField` class. It can share its styling with `vTextField`. mackong, do you want to implement that in this PR? This is more of a cleanup/optimization than a "new feature" so a release note isn't needed.

**Pull request merged**